helix-view/src/view.rs

```diff
@@ -104,23 +109,43 @@ impl View {
 
         let last_col = self.offset.col + inner_area.width.saturating_sub(1) as usize;
 
-        if line > last_line.saturating_sub(scrolloff) {
+        let row = if line > last_line.saturating_sub(scrolloff) {
             // scroll down
-            self.offset.row += line - (last_line.saturating_sub(scrolloff));
+            self.offset.row + line - (last_line.saturating_sub(scrolloff))
         } else if line < self.offset.row + scrolloff {
             // scroll up
-            self.offset.row = line.saturating_sub(scrolloff);
-        }
+            line.saturating_sub(scrolloff)
+        } else {
+            self.offset.row
+        };
 
-        if col > last_col.saturating_sub(scrolloff) {
+        let col = if col > last_col.saturating_sub(scrolloff) {
             // scroll right
-            self.offset.col += col - (last_col.saturating_sub(scrolloff));
+            self.offset.col + col - (last_col.saturating_sub(scrolloff))
         } else if col < self.offset.col + scrolloff {
             // scroll left
-            self.offset.col = col.saturating_sub(scrolloff);
+            col.saturating_sub(scrolloff)
+        } else {
+            self.offset.col
+        };
+        if row == self.offset.row && col == self.offset.col {
+            None
+        } else {
+            Some((row, col))
         }
     }
 
+    pub fn ensure_cursor_in_view(&mut self, doc: &Document, scrolloff: usize) {
+        if let Some((row, col)) = self.offset_coords_to_in_view(doc, scrolloff) {
+            self.offset.row = row;
+            self.offset.col = col;
+        }
+    }
+
+    pub fn is_cursor_in_view(&mut self, doc: &Document, scrolloff: usize) -> bool {
+        self.offset_coords_to_in_view(doc, scrolloff).is_none()
+    }
```

Comment:
Do we really need to call `offset_coords_to_in_view` twice?

Reply:
Scrolloff is different between two calls.
I don't have idea to merge two calls.